### PR TITLE
Update button focus prop

### LINKED ISSUE #50972: Button: Undocument the `focus` prop
## What problem does this address?

The documentation for the Button component currently advertises a [`focus` prop](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/button/README.md#focus-boolean), which as far as I can tell, has only been working for a four-month period in 2018. ([First implemented](https://github.com/WordPress/gutenberg/pull/6527/files#diff-47628eb6e63618c43a7afb648c04426d01b0ed4ec5e3bb79fb820df1ba4f62baR47) in May 2018 as an alias for the `autoFocus` prop, and then [removed](https://github.com/WordPress/gutenberg/pull/9572/files#diff-1204d0def76e28ff29950465cc842e3fafcee022c7e97c76fa5cf5baa4c4505aL48) without explanation four months later.)

## What is your proposed solution?

Given that this has barely been working in production, and its originally intended behavior can be accomplished via the standard `autoFocus` prop, I suggest that we simply remove this from the documentation (readme and TypeScript types).

### PR DESCRIPTION
**Pull request by noahbcaldwell**
## What?

Updating button focus prop in documentation

Closes #50972

## Why?

Button focus prop has been removed

### CODE REVIEW

File: packages/components/src/button/types.ts [hunk 1]
```diff
@@ -45,10 +45,6 @@ type BaseButtonProps = {
 	/**
 	 * Whether the button is focused.
 	 */
-	focus?: boolean;
-	/**
-	 * If provided, renders an Icon component inside the button.
-	 */
```
Comment (ciampo) [inline packages/components/src/button/types.ts:51]: This description text refers to the `icon` prop, and should not be deleted. We should detele, instead, the lines above the prop ("Whether the button is focused." text)

**Review by ciampo — COMMENTED** [review]
Thank you for opening this PR!

I've tweaked the PR description to link to the original issue.

Also, I can see changes to a few files that should not be part of this PR — could you remove them? 

- `.vscode/settings.json`
- `packages/scripts/CHANGELOG.md`
- `packages/scripts/README.md`

Thank you!

File: packages/components/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### Documentation
+
+-   Button `focus` prop: Update documentation to remove the button `focus` prop ([#50972](https://github.com/WordPress/gutenberg/issues/50972)).
+
```
Comment (chad1008) [inline packages/components/CHANGELOG.md:8]: Looks like the PR number needs to be updated here (to 51808)

**Review by chad1008 — COMMENTED** [review]
Thanks for the PR 🎉 

I made one small note on the CHANGELOG entry, but that's all from me, other than @ciampo's previous notes

**Comment by mburridge** [comment]
@noahbcaldwell just nudging this following @ciampo's [suggestion](https://github.com/WordPress/gutenberg/pull/51808#pullrequestreview-1494613575) to remove unnecessarily changed files.

**Comment by ciampo** [comment]
@noahbcaldwell are you still able to work on this PR? We are going to take over, otherwise.

**Comment by noahbcaldwell** [comment]
> @noahbcaldwell are you still able to work on this PR? We are going to take over, otherwise.

So sorry for the late reply! Life has been hectic. Unfortunately I am not available to work on this PR right now. Feel free to assign it to someone else!

**Comment by ciampo** [comment]
@chad1008 would you be able to wrap this up?

**Comment by chad1008** [comment]
No problem @ciampo!

**Comment by chad1008** [comment]
Rather than tangle with the differing remotes due to this PR coming from a fork, I'm going to close this one in favor of #54397. Thanks for getting the ball rolling @noahbcaldwell!

**Pull request closed without merging**